### PR TITLE
perf(compaction-test): improve the performance of deterministic compaction test tool

### PR DESCRIPTION
**Pull request by StrikeW**
I hereby agree to the terms of the [Singularity Data, Inc. Contributor License Agreement](https://gist.github.com/skyzh/0663682a70b0edde7ae991492f2314cb#file-s9y_cla).

## What's changed and what's your intention?

1. Use iterator to fetch kv pairs
2. Spawn a task to check the correctness of compaction results

The tool can replay 378 version deltas and 10GB L0 SSTs in about 2 mins with risingwave built in release build.

## Checklist

- [x] I have written necessary rustdoc comments
~- [ ] I have added necessary unit tests and integration tests~
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)

## Documentation

If your pull request contains user-facing changes, please specify the types of the changes, and create a release note. Otherwise, please feel free to remove this section.

### Types of user-facing changes

Please keep the types that apply to your changes, and remove those that do not apply.

* Installation and deployment 
* Connector (sources & sinks)
* SQL commands, functions, and operators
* RisingWave cluster configuration changes
* Other (please specify in the release note below)

### Release note

Please create a release note for your changes. In the release note, focus on the impact on users, and mention the environment or conditions where the impact may occur.

## Refer to a related PR or issue link (optional)
#5441

### CODE REVIEW

File: src/storage/compaction_test/src/server.rs [hunk 1]
```diff
@@ -43,8 +46,15 @@ struct Metrics {
...
 
-/// This tool will be started after we generate enough L0 SSTs to Hummock.
-/// Fetches and runs compaction tasks.
+/// Steps to use the compaction test tool
+///
+/// 1. Start the cluster with full-compaction-test config: `./risedev d full-compaction-test`
+/// 2. Ingest enough L0 SSTs, for example we can use the tpch-bench tool
+/// 3. Disable hummock manager commit new epochs: `./risedev ctl hummock disable-commit-epoch`,
+/// and it will print the current max committed epoch in Meta. Note that we need to wait for the
+/// frontend to unpin that epoch before going to next step.
```
Comment (zwang28) [inline src/storage/compaction_test/src/server.rs:55]: We may query status of epoch pin by rise_ctl_get_pinned_snapshots_summary.

Comment (zwang28) [reply src/storage/compaction_test/src/server.rs:55]: Or how can we know "the frontend has unpinned that epoch before going to next step"?

Comment (StrikeW) [reply src/storage/compaction_test/src/server.rs:55]: I think currently we can use a poll-based method via `rise_ctl_get_pinned_snapshots_summary ` as you suggested to check that.

File: src/storage/compaction_test/src/server.rs [hunk 2]
```diff
@@ -185,18 +195,19 @@ pub async fn compaction_test_serve(
...
 
             local_version_manager.try_update_pinned_version(
                 pin_version_response::Payload::PinnedVersion(new_version),
             );
 
-            check_compaction_results(&expect_results, &hummock, opts.table_id).await?;
+            let new_version_iters = open_hummock_iters(&hummock, &epochs, opts.table_id).await?;
 
+            // spawn a task to check the results
+            tokio::spawn(check_compaction_results(
```
Comment (zwang28) [inline src/storage/compaction_test/src/server.rs:206]: Should we join these task to ensure they are finished?

Comment (StrikeW) [reply src/storage/compaction_test/src/server.rs:206]: Good catch.

File: src/storage/compaction_test/src/server.rs (hunk 1 above)
Comment (StrikeW) [reply src/storage/compaction_test/src/server.rs:55]: I plan to submit the following PR to fix these flaws. This PR focues on performance improvement.

**Comment by StrikeW** [comment]
Duplicated with #5919

**Pull request closed without merging**